### PR TITLE
feat: add preset event _page_load

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
## Description
1.  add preset event _page_load

## General Checklist

- [x] Added new tests to cover change, if needed
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/tracker/PageLoadTracker.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			}
+			this.provider.record({
+				name: Event.PresetEvent.PAGE_LOAD,
+				attributes: eventAttributes,
+			});
+		}
+	}
+
+	isPageLoaded() {
+		const performanceEntries = performance.getEntriesByType('navigation');
```
Comment (chenhaiyun) [inline src/tracker/PageLoadTracker.ts:60]: Should add a condition to check the performance is working?

File: src/tracker/PageLoadTracker.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			const latestPerformance =
+				performanceEntries[performanceEntries.length - 1];
+			const eventAttributes: ClickstreamAttribute = {};
+			for (const key in latestPerformance) {
+				const value = (latestPerformance as any)[key];
+				const valueType = typeof value;
+				if (Event.ReservedAttribute.TIMING_ATTRIBUTES.includes(key)) {
+					if (valueType === 'string' || valueType === 'number') {
+						eventAttributes[key] = value;
+					} else if (Array.isArray(value) && value.length > 0) {
```
Comment (chenhaiyun) [inline src/tracker/PageLoadTracker.ts:47]: Do we need add the `object` type in this condition.

File: src/tracker/PageLoadTracker.ts (hunk 1 above)
Comment (zhu-xiaowei) [reply src/tracker/PageLoadTracker.ts:60]: Which have already checked at the beginning of [init method](https://github.com/awslabs/clickstream-web/pull/40/files/2ed40838925456a421a5e3a84ea690cceccbb7f1#diff-fe1b57e22b0ad16aba11da9f10fea3977f70349f91f8b04cdab7fa888de5967aR23)

File: src/tracker/PageLoadTracker.ts (hunk 2 above)
Comment (zhu-xiaowei) [reply src/tracker/PageLoadTracker.ts:47]: Clickstream SDK do not support object type in custom attributes, and here we currently only support the filtered preset [timing attribute](https://github.com/awslabs/clickstream-web/pull/40/files/2ed40838925456a421a5e3a84ea690cceccbb7f1#diff-812f995ec9b9838ad0eea509bcaa131f10e718ff5a665069c85bf6e3d244d897R73) which only contains `string`, `number` and `array` type.

File: src/tracker/PageLoadTracker.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+	init() {
+		this.trackPageLoad = this.trackPageLoad.bind(this);
+		if (this.isSupportedEnv()) {
+			this.observer = new PerformanceObserver(() => {
+				this.trackPageLoad();
+			});
+			this.observer.observe({ entryTypes: ['navigation'] });
+		}
+		if (this.isPageLoaded()) {
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply src/tracker/PageLoadTracker.ts:29]: PerformanceNavigationTiming event will assigned all timing attributes after the window `load`（loadEventEnd） lifecycle:
[image](https://developer.mozilla.org/en/docs/Web/API/PerformanceNavigationTiming/timestamp-diagram.svg)
and after the loadEventEnd, the timing attribute `duration` will be assigned last. Here we judge that if the [duration](https://github.com/awslabs/clickstream-web/pull/40/files#diff-fe1b57e22b0ad16aba11da9f10fea3977f70349f91f8b04cdab7fa888de5967aR61) has been assigned a positive value during initialization of the SDK, it proves that the page has been loaded, and the `_page_load` time will be sent.

SPA page is the same logic, we will track `_page_load` event when the SDK is initialized or the PerformanceObserver triggered to ensure that `_page_load` event is only triggered once with all timing attributes.

**Review by chenhaiyun — APPROVED** [review]

**Pull request merged**